File: Containers/ListStack.py
import LinkedList


class ListStack:
    def __init__(self):
        self.linkedList = LinkedList.LinkedList()

    def isEmpty(self):
        return self.linkedList.isEmpty()

    def top(self):
        if not self.linkedList.isEmpty():
            return self.linkedList.head().value

    def pop(self):
        if not self.linkedList.isEmpty():
            t = self.linkedList.head()
            self.linkedList.remove(t)
            return t.value

    def push(self, v):
        self.linkedList.insert(self.linkedList.head(), v)


# listStack = ListStack()
# listStack.push(1)
# listStack.push(2)
# listStack.push(3)
# listStack.push(4)
# listStack.pop()
# listStack.pop()
# print(listStack.top())

File: Containers/ListStack.java
package Containers;

public class ListStack<T> {
    private LinkedList<T> linkedList;

    public ListStack() {
        linkedList = new LinkedList<>();
    }

    public boolean isEmpty() {
        return linkedList.isEmpty();
    }

    public T top() {
        if (linkedList.isEmpty())
            throw new IllegalStateException("Stack is empty");
        return linkedList.head().value;
    }

    public T pop() {
        if (linkedList.isEmpty())
            throw new IllegalStateException("Stack is empty");
        Node<T> t = linkedList.head();
        linkedList.remove(t);
        return t.value;
    }

    public void push(T value) {
        linkedList.append(value);
    }
}
